### PR TITLE
Material - Extra fields

### PR DESCRIPTION
**Pull request by Moo64c**
#16

### CODE REVIEW

File: garmentbox/modules/custom/garmentbox_material/plugins/content_types/material_items/material_items.inc [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  if (empty($result['field_collection_item'])) {
+    // None of the BOM field collections point to the material.
+    return;
+  }
+  $bom_collection_ids = array_keys($result['field_collection_item']);
+
+  // Find item variants that use the material.
+  $query = new EntityFieldQuery();
+  $result = $query
```
Comment (bricel) [inline garmentbox/modules/custom/garmentbox_material/plugins/content_types/material_items/material_items.inc]: Remove

File: garmentbox/themes/garmentbox_omega/template.php [hunk 2]
```diff
@@ -28,7 +28,55 @@ function garmentbox_omega_preprocess_page(&$variables) {
...
+ */
+function garmentbox_omega_preprocess_material_node_garmentbox_header(&$variables) {
+  // Add variables to create the "Add material" button.
+  $variables['theme_hook_suggestions'][] = 'node__header';
+
+  $content = &$variables['content'];
+  $content['add_material_link'] = l('Add new material', 'node/add/material');
+
+  // Display title instead of nickname.
+  $content['field_nick_name'][0]['#markup'] = $variables['title'];
```
Comment (amitaibu) [inline garmentbox/themes/garmentbox_omega/template.php]: XSS

File: garmentbox/modules/custom/garmentbox_bom/garmentbox_bom.module [hunk 3]
```diff
@@ -135,11 +135,11 @@ function garmentbox_bom_get_variant_bom_price($node) {
 function garmentbox_bom_get_material_price($collection_item) {
   $wrapper = entity_metadata_wrapper('field_collection_item', $collection_item);
 
-  $m_unit = $wrapper->field_material_item->field_unit->value();
-  $m_length_unit = $wrapper->field_material_item->field_length_unit->value();
-  $m_length_unit_conversion = ($wrapper->field_material_item->field_length_unit->value() && !empty($wrapper->field_material_item->field_length_unit->field_conversion_ratio)) ? $wrapper->field_material_item->field_length_unit->field_conversion_ratio->value() : 1;
-  $m_length = $wrapper->field_material_item->field_length->value();
-  $m_price = $wrapper->field_material_item->field_price->value();
+  $m_unit = $wrapper->field_source_info_ref->field_unit->value();
```
Comment (amitaibu) [inline garmentbox/modules/custom/garmentbox_bom/garmentbox_bom.module]: Make variable more descriptive. 

$m_unit -> $material_unit
...

File: garmentbox/modules/custom/garmentbox_material/plugins/content_types/material_items/material_items.inc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  if (empty($result['field_collection_item'])) {
+    // None of the BOM field collections point to the material.
+    return;
+  }
+  $bom_collection_ids = array_keys($result['field_collection_item']);
+
+  // Display a view.
+  $view = views_get_view('material');
+  $view->set_arguments(array(implode(',',$bom_collection_ids)));
```
Comment (amitaibu) [inline garmentbox/modules/custom/garmentbox_material/plugins/content_types/material_items/material_items.inc]: Missing space before the `$bom_collection_ids`

File: garmentbox/modules/custom/garmentbox_material/plugins/content_types/material_items/material_items.inc [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  $output = $view->preview('material_item_variants');
```
Comment (amitaibu) [inline garmentbox/modules/custom/garmentbox_material/plugins/content_types/material_items/material_items.inc]: No need for $output, assign it directly to $block->content.

File: garmentbox/themes/garmentbox_omega/template.php [hunk 6]
```diff
@@ -28,7 +28,55 @@ function garmentbox_omega_preprocess_page(&$variables) {
...
+
+/**
+ * Material node preprocess.
+ *
+ * @see garmentbox_omega_preprocess_node().
+ */
+function garmentbox_omega_preprocess_material_node(&$variables) {
+  $content = &$variables['content'];
+
+  $hidden_fields = array('og_group_ref', 'body', 'field_images', 'field_nick_name', 'field_source_info', 'field_material_type');
```
Comment (amitaibu) [inline garmentbox/themes/garmentbox_omega/template.php]: That's really hardcoded -- why do we need this? Why not `hook_field_access()`?

File: garmentbox/behat/features/material.feature [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    And I should see a table titled "Task list" with the following <contents>:
+      | Summary  | Status     | Assignee | Replies | Last updated | Created  | Actions |
+      | Buy more | Needs work |          | 0       | <ignore>     | <ignore> | edit    |
+      | Clean    | Open       |          | 0       | <ignore>     | <ignore> | edit    |
+    And I should see a table titled "Source info" with the following <contents>:
+      | Vendor               | Price  | Length conversion | Unit     | Length unit | Operations  |
+      | Vogue Fabrics        | $10.00 | 20.00             | Kilogram | Meter       | Edit Delete |
+      | Vogue Fabrics        | $20.00 | 10.00             | Kilogram | Meter       | Edit Delete |
+      | High Fashion Fabrics | $30.00 | 10.00             | Kilogram | Meter       | Edit Delete |
+    And I should see "Total open tasks: 2"
```
Comment (amitaibu) [inline garmentbox/behat/features/material.feature]: Should be `And I should see the following:`

File: garmentbox/modules/custom/garmentbox_material/garmentbox_material.module [hunk 8]
```diff
@@ -88,3 +98,34 @@ function garmentbox_material_get_units($type) {
...
   return !$result ? array() : array_keys($result['taxonomy_term']);
 }
+
+/**
+ * Implements hook_field_access().
+ *
+ * Hide fields for material node view.
+ */
+function garmentbox_material_field_access($op, $field, $entity_type, $entity, $account) {
+  if ($op != 'view' || empty($entity->type) || $entity->type != 'material') {
```
Comment (amitaibu) [inline garmentbox/modules/custom/garmentbox_material/garmentbox_material.module:108]: Remove empty($entity->type). Maybe replace with `$entity_type != 'node'`

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**